sycl/source/detail/device_image_impl.hpp

```diff
@@ -142,6 +142,9 @@ class ManagedDeviceBinaries {
   sycl_device_binaries MBinaries;
 };
 
+using MangledKernelNameMapT = std::map<std::string, std::string, std::less<>>;
```

Comment:
I want to highlight this part to the reviewers: this container used to be an `std::unordered_map`, but those don't support heterogenous lookup in C++17. 

An alternative to this can be using a separate container for storing strings while using this one to work with string views. This would require special handling when merging multiple images, so I'm leaning towards leaving that as a potential future optimization, but let me know if you feel otherwise.